### PR TITLE
Custom initial day of week

### PR DESCRIPTION
**Pull request by gsrathoreniks**

### CODE REVIEW

**Comment by gsrathoreniks** [comment]
Closes #15

File: kalendar-endlos/src/main/java/com/himanshoe/kalendar/endlos/ui/KalendarMonth.kt [hunk 1]
```diff
@@ -94,17 +95,28 @@ internal fun KalendarMonth(
...
     return mutableListOf<LocalDate>().apply {
         val firstDay = monthState.value.atDay(1)
-        val firstSunday = if (firstDay.dayOfWeek == java.time.DayOfWeek.SUNDAY) {
+        val firstDayOfWeek = if (firstDay.dayOfWeek == kalendarKonfig.firstDayOfWeek) {
             firstDay
         } else {
-            firstDay.minusDays(firstDay.dayOfWeek.value.toLong())
+            firstDay.minusDays(firstDay.dayOfWeek.value - kalendarKonfig.firstDayOfWeek.value.toLong())
+        }
+        val weekList = listOf(7, 1, 2, 3, 4, 5, 6)
```
Comment (hi-manshu) [inline kalendar-endlos/src/main/java/com/himanshoe/kalendar/endlos/ui/KalendarMonth.kt:106]: Move this out to global private variable

File: kalendar-endlos/src/main/java/com/himanshoe/kalendar/endlos/ui/KalendarMonth.kt [hunk 2]
```diff
@@ -94,17 +95,28 @@ internal fun KalendarMonth(
...
+            firstDay.dayOfWeek.value > kalendarKonfig.firstDayOfWeek.value -> {
+                val moves = firstDay.dayOfWeek.value - kalendarKonfig.firstDayOfWeek.value
+                Collections.rotate(weekList, moves)
+            }
+            firstDay.dayOfWeek.value > kalendarKonfig.firstDayOfWeek.value -> {
+                val moves = kalendarKonfig.firstDayOfWeek.value - firstDay.dayOfWeek.value
+                Collections.rotate(weekList, moves)
+            }
         }
         repeat(6) { weekIndex ->
```
Comment (hi-manshu) [inline kalendar-endlos/src/main/java/com/himanshoe/kalendar/endlos/ui/KalendarMonth.kt:117]: Can you move out constants please?

File: kalendar-endlos/src/main/java/com/himanshoe/kalendar/endlos/ui/KalendarMonth.kt [hunk 3]
```diff
@@ -94,17 +95,28 @@ internal fun KalendarMonth(
...
             (0..6).forEach { dayIndex ->
-                add(firstSunday.plusDays((7 * weekIndex + dayIndex).toLong()))
+                add(firstDayOfWeek.plusDays((7 * weekIndex + dayIndex).toLong()))
```
Comment (hi-manshu) [inline kalendar-endlos/src/main/java/com/himanshoe/kalendar/endlos/ui/KalendarMonth.kt:119]: in plus of + use add

File: kalendar-endlos/src/main/java/com/himanshoe/kalendar/endlos/ui/KalendarWeekDayNames.kt [hunk 4]
```diff
@@ -37,14 +37,15 @@ import androidx.compose.ui.Modifier
...
+import java.util.Collections
 
 private const val DAYS_IN_WEEK = 7
 private const val ZERO = 0
 
 @Composable
 internal fun KalendarWeekDayNames(kalendarKonfig: KalendarKonfig) {
     val weekdays = DateFormatSymbols(kalendarKonfig.locale).weekdays.takeLast(DAYS_IN_WEEK)
-
+    Collections.rotate(weekdays, 7 - kalendarKonfig.firstDayOfWeek.value)
```
Comment (hi-manshu) [inline kalendar-endlos/src/main/java/com/himanshoe/kalendar/endlos/ui/KalendarWeekDayNames.kt:48]: use minus

File: kalendar-endlos/src/main/java/com/himanshoe/kalendar/endlos/ui/KalendarMonth.kt (hunk 2 above)
Comment (gsrathoreniks) [reply kalendar-endlos/src/main/java/com/himanshoe/kalendar/endlos/ui/KalendarMonth.kt:117]: done

**Pull request merged**